engine/src/main/java/org/terasology/particles/components/ParticleSystemComponent.java

```diff
@@ -0,0 +1,155 @@
+
+    /**
+     * Destroy this particle system when its last emitter is destroyed TODO: Unimplemented
+     */
+    public boolean destroyWhenFinished = true;
+
+    /**
+     * The amount of particles in this system's pool. Emitters under this system will share the particles.
+     */
+    public int particlePoolSize = 1000;
+
+    @Owns
+    private List<EntityRef> emitters = new LinkedList<>();
+
+    @Owns
+    private List<EntityRef> affectors = new LinkedList<>();
+
+    /**
+     * This system's particle texture
+     */
+    public Texture texture = null;
+
+    /**
+     * This system's particle texture size, in percents x: [0.0, 1.0], y: [0.0, 1.0]
+     */
+    public Vector2f textureSize = new Vector2f(1.0f, 1.0f);
+
+
+    /**
+     * Notifies ParticleSystemManagerImpl that this system needs an updated ParticleSystemStateData
+     */
+    private void requestUpdate() {
+        if (ownerEntity != null) {
+            ownerEntity.send(new ParticleSystemUpdateEvent());
+        }
+    }
+
+    /**
+     * Changes the particle system's desired renderer. Will register a new renderer if it does not already exist.
+     *
+     * @param renderer The renderer to set this system to.
+     */
+    public void setRenderer(Class<ParticleRenderer> renderer) {
+        this.renderer = renderer;
+        requestUpdate();
+    }
+
+    /**
+     * @return The renderer this system should be rendered with.
+     */
+    public Class<? extends ParticleRenderer> getRenderer() {
+        return renderer != null ? renderer : org.terasology.particles.rendering.DisplayListParticleRenderer.class;
+    }
+
+    // EMITTER LIST ACCESS
+
+    public void addEmitter(final EntityRef emitter) {
+        emitters.add(emitter);
+        requestUpdate();
+    }
+
+    public boolean removeEmitter(final EntityRef emitter) {
+        boolean removed = emitters.remove(emitter);
+        requestUpdate();
+        return removed;
+    }
+
+    public EntityRef removeEmitter(final int emitterIndex) {
+        EntityRef emitter = emitters.remove(emitterIndex);
+        requestUpdate();
+        return emitter;
+    }
+
+    public EntityRef getEmitter(int index) {
+        return emitters.get(index);
+    }
+
+    public List<EntityRef> getEmitters() {
+        return emitters;
+    }
+
+    // AFFECTOR LIST ACCESS
+
+    public void addAffector(final EntityRef affector) {
+        affectors.add(affector);
+        requestUpdate();
+    }
+
+    public boolean removeAffector(final EntityRef affector) {
+        boolean removed = affectors.remove(affector);
+        requestUpdate();
+        return removed;
+    }
+
+    public EntityRef removeAffector(final int affectorIndex) {
+        EntityRef affector = affectors.remove(affectorIndex);
+        requestUpdate();
+        return affector;
+    }
+
+    public EntityRef getAffector(int index) {
+        return affectors.get(index);
+    }
+
+    public List<EntityRef> getAffectors() {
+        return affectors;
+    }
```

Comment:
Looking back on this class, I'm not convinced we need it. 

We need ParticleDataComponents, ParticleEmitterComponents, ParticleRenderingComponents, ParticleAffectorsComponets but what glues them together should be specialized systems i.e. CampFireSystem and probably an overseeing ParticleManager more or less like you already have.